### ISSUE TITLE
Feature Request: Hausdorff distance

### ISSUE THREAD

**Issue by jakubMitura14** [pytorch/pytorch#73778]
### 🚀 The feature, motivation and pitch

I developed CUDA C++ algorithm for calculating Hausdorff distance for 3 dimensional arrays. Algortihm is developed for evaluation of medical image segmentation, but can be adapted to other 3D usecases, and preliminary benchmarks shown it is probably the fastest implementation based on mathemathical morphology ( 10 times faster then previous fastest developed by prof Oliviera  [1] - and at least  two orders of magnitute faster  than CPU implementation ).
I have classic, robust and average Hausdorff distance. 
There is good article not mine [2]  giving possibilities related to Hausdorff distance and How it can be used in convolutional neural networks.

 I would like to make it available to larger amount of researchers and developers, I consider contributing either to MONAI or to Pytorch, depending on feedback I will get here.

1) https://www.sciencedirect.com/science/article/abs/pii/S003132032100176X
2) https://arxiv.org/abs/1904.10030

**Comment by jakubMitura14** [comment]
Yes, I suppose I did not made my statement clear I do not state that I want to add unique new metric - I state My algorithm is probably most efficient algorithm to calculate Hausdorff distance.

**Comment by jakubMitura14** [comment]
Ok thanks for response :)
